query/src/users/auth/jwt/authenticator.rs

```diff
@@ -35,18 +35,26 @@ pub struct JwtAuthenticator {
 
 #[derive(Default, Deserialize, Serialize)]
 pub struct EnsureUser {
-    pub tenant_id: Option<String>,
     pub roles: Vec<String>,
 }
 
 #[derive(Default, Deserialize, Serialize)]
 pub struct CustomClaims {
+    pub tenant_id: Option<String>,
     pub ensure_user: Option<EnsureUser>,
 }
 
 impl CustomClaims {
     pub fn new() -> Self {
-        CustomClaims { ensure_user: None }
+        CustomClaims {
+            tenant_id: None,
+            ensure_user: None,
+        }
+    }
+
+    pub fn with_tenant_id(mut self, tenant_id: &str) -> Self {
```

Comment:
how about move a `tenant_id: String` here? maybe this could reduce a string copy

Reply:
This is for testing only.